test/test-viz.cpp

```diff
@@ -0,0 +1,350 @@
+#define SDL_MAIN_HANDLED
+
+//Standard libs includes
+#include <iostream>
+#include <vector>
+#include <string>
+#include <tuple>
+#include <variant>
+
+//External libs includes
+#include <lorina/bench.hpp>
+#include <SDL.h>
+
+#define SCREEN_WIDTH 1920
+#define SCREEN_HEIGHT 1080
+
+#define HOR_SPACING 2
+#define VER_SPACING 2
+
+struct Vertex {
+    std::string type;
+    std::string name;
+    int layer;
+    std::vector<Vertex> ins;
+    SDL_FRect position;
+};
+
+struct Line {
+    double x1;
+    double y1;
+    double x2;
+    double y2;
+};
+
+class logic_scheme {
+public:
+
+    logic_scheme() {
+        _outputNum = 0;
+    }
+
+    void add_input(const std::string& _name) {
+        Vertex input;
+        input.type = "IN";
+        input.name = _name;
+        input.layer = 0;
+
+        _scheme.emplace_back(input);
+    }
+
+    void add_output(const std::string& _input) {
+        Vertex output;
+        output.type = "OUT";
+        output.name = std::string("output" + std::to_string(_outputNum));
+
+        _outputNum++;
+
+        std::vector<std::string> ins;
+        ins.emplace_back(_input);
+
+        _queue.emplace_back(output, ins);
+
+        _scheme.emplace_back(output);
+    }
+
+    void add_dff(const std::string& _input, const std::string& _output) {
+        Vertex dff;
+
+        dff.type = "DFF";
+        dff.name = _output;
+
+        std::vector<std::string> ins;
+        ins.emplace_back(_input);
+
+        if(check_inputs(dff, ins)) _queue.emplace_back(dff, ins);
+
+        _scheme.emplace_back(dff);
+    }
+
+    void add_gate(const std::vector<std::string>& _inputs, const std::string& _output, const std::string& _type) {
+        Vertex gate;
+        gate.type = _type;
+        gate.name = _output;
+        std::vector<std::string> ins = _inputs;
+
+        if(check_inputs(gate, ins)) _queue.emplace_back(gate, ins);
+
+        _scheme.emplace_back(gate);
+    }
+
+    //Method to check whether demanded input exists or not
+    bool check_inputs(Vertex& element, std::vector<std::string>& ins) {
```

Comment:
The naming is ambiguous: what is the result of the check? Moreover, method body does something very different to just checking inputs.